doc: fix structure issues in some libs + minor doc fixes (#7403)
nrfconnect/sdk-nrf: doc/nrf/libraries/networking/aws_iot.rst

.. note::
   The AWS IoT library and the :ref:`lib_aws_fota` library share common steps related to `AWS IoT console`_ management.

Requirements
************

> To use this library, you must build and program the :ref:`AWS FOTA <aws_fota_sample>` sample, available for the nRF9160 DK.

Reviewer: ```suggestion
To use this library, you must build and program the :ref:`AWS FOTA <aws_fota_sample>` sample available for the nRF9160 DK.
```

Reviewer 2: ```suggestion
To use this library, you can build and program the :ref:`AWS IoT <aws_iot>` sample, available for the nRF9160 DK.
```

Reviewer: This does not sound right. My guess is something to this direction:
"You must enable this library to be able to use (or build and program?) the :ref:`aws_iot` sample available for..."

And it still feels a bit weird, because we usually state something about enabling a library in the sample docs. Not like here. The AWS IoT sample already mentions that you would need to enable this library to be able to build and program the sample.

Reviewer 2: Yeah..may be we do not need this at all here?

Author: Maybe "This library is meant for use with the <sample>"?

Reviewer 2: @simensrostad could you suggest?

Reviewer 3: Yes. The idea is that the AWS IoT sample showcases the use of the AWS IoT library.

> As part of this process, you will :ref:`create a thing in AWS IoT <creating_a_thing_in_AWS_IoT>` and generate certificates for the TLS connection.

Reviewer: ```suggestion
As part of this process, you :ref:`create a thing in AWS IoT <creating_a_thing_in_AWS_IoT>` and generate certificates for the TLS connection.
```

.. _set_up_conn_to_iot:

Configuration
*************

> After meeting the `Requirements`_, set up a secure MQTT connection to the AWS IoT message broker by completing the following configuration procedures:

Reviewer: ```suggestion
After meeting the `Requirements`_, complete the following configuration procedures to set up a secure MQTT connection to the AWS IoT message broker:
```

* Programming the certificates to the on-board modem of the nRF9160-based kit.
* Configuring the required library options.

> * Configuring the optional library options.

Reviewer: Maybe imperative instead of gerund?

.. note::
   The AWS IoT library and the :ref:`lib_aws_fota` library share common steps related to `AWS IoT console`_ management.

Requirements
************

> This library is meant for use with the :ref:`AWS FOTA <aws_fota_sample>` sample available for the nRF9160 DK.
> As part building the sample, you :ref:`create a thing in AWS IoT <creating_a_thing_in_AWS_IoT>` and generate certificates for the TLS connection.

Reviewer 2: I am thinking now if this should be part of just the section above and not a requirement.
```suggestion
The :ref:`aws_iot` sample available for the nRF9160 DK showcases the use of this library.
As part of building the sample, you :ref:`create a thing in AWS IoT <creating_a_thing_in_AWS_IoT>` and generate certificates for the TLS connection.
```

.. note::
   The default security tag set by the :guilabel:`Certificate manager` *16842753* is reserved for communications with :ref:`lib_nrf_cloud`.

Configure the required library options
======================================

Complete the following steps:

1. Set the following options to establish a connection to the AWS IoT message broker:

   * :kconfig:option:`CONFIG_AWS_IOT_SEC_TAG`
   * :kconfig:option:`CONFIG_AWS_IOT_BROKER_HOST_NAME`
   * :kconfig:option:`CONFIG_AWS_IOT_CLIENT_ID_STATIC`

#. Complete the following steps to set the required library options:

   a. In the `AWS IoT console`_, navigate to :guilabel:`IoT core` > :guilabel:`Settings`.
   #. Find the ``Endpoint`` address and set :kconfig:option:`CONFIG_AWS_IOT_BROKER_HOST_NAME` to this address string.
   #. Set the option :kconfig:option:`CONFIG_AWS_IOT_CLIENT_ID_STATIC` to the name of the *thing* created earlier in the process.

>    #. Set the security tag configuration :kconfig:option:`CONFIG_AWS_IOT_SEC_TAG` to the security tag, chosen while `Programming the certificates to the on-board modem of the nRF9160-based kit`_.

Reviewer: ```suggestion
   #. Set the security tag configuration :kconfig:option:`CONFIG_AWS_IOT_SEC_TAG` to the security tag, chosen while you `Program the certificates to the on-board modem of the nRF9160-based kit`_.
```

Configuring the required library options
========================================

Complete the following steps:

> 1. Establish a connection to the AWS IoT message broker by setting the following options:

Reviewer: ```suggestion
1. Set the following options to establish a connection to the AWS IoT message broker:
```

   * :kconfig:option:`CONFIG_AWS_IOT_SEC_TAG`
   * :kconfig:option:`CONFIG_AWS_IOT_BROKER_HOST_NAME`
   * :kconfig:option:`CONFIG_AWS_IOT_CLIENT_ID_STATIC`

> #. Configure the required library options by completing the following steps:

Reviewer: ```suggestion
#. Complete the following steps to set the required library options:
```

   a. In the `AWS IoT console`_, navigate to :guilabel:`IoT core` > :guilabel:`Settings`.

>    #. Find the ``Endpoint`` address and set the configurable option :kconfig:option:`CONFIG_AWS_IOT_BROKER_HOST_NAME` to this address string.

Reviewer: ```suggestion
   #. Find the ``Endpoint`` address and set :kconfig:option:`CONFIG_AWS_IOT_BROKER_HOST_NAME` to this address string.
```

>    #. Set the option :kconfig:option:`CONFIG_AWS_IOT_CLIENT_ID_STATIC` to the name of the *thing* created during the aforementioned steps.

Reviewer: ```suggestion
   #. Set the option :kconfig:option:`CONFIG_AWS_IOT_CLIENT_ID_STATIC` to the name of the *thing* created earlier in the process.
```

   To enable the use of the cloud, API set the configurable option :kconfig:option:`CONFIG_CLOUD_API`, in addition to the other selected library options.

> Initializing the library

Reviewer 2: Line 98, since it just one Kconfig option, we could combine it with the sentence and avoid the bullet point.

========================

The library is initialized by calling the :c:func:`aws_iot_init` function.
If this API call fails, the application must not make any other API calls to the library.

Connecting to the AWS IoT message broker
========================================

After the initialization, the :c:func:`aws_iot_connect` function must be called to connect to the AWS IoT broker.
If this API call fails, the application must retry the connection by calling :c:func:`aws_iot_connect` again.

.. note::

>    The connection attempt can fail due to any number of reasons related to external network.

Reviewer: ```suggestion
   The connection attempt can fail due to a number of reasons related to external network.
```
